arangod/Cluster/FollowerInfo.h

```diff
@@ -197,6 +197,10 @@ class FollowerInfo {
 #endif
         return _canWrite;
       }
+      if (!_theLeaderTouched) {
```

Comment:
Writes to _theLeaderTouched are not protected by the locks.
However this can only go from false to true, not back.
So if we are unlucky we might reject the document this is being written concurrently to taking over the leadership.

I think this is acceptable, and should in general be better than adding locking around theLeaderUntouched.
Also we do not have undefined behavior here.

Reply:
I agree, although we could put this check here under the `_dataLock` to be entirely sure. I doubt this would cost much due to the `_canWrite` tuning.

Reply:
As I understand it, that is actually undefined behaviour. http://eel.is/c++draft/intro.races#21

If it isn't too costly, I wouldn't invest in thinking and arguing whether it could be safe in practice. So I'd be in favour of doing this either under a lock, or as atomic operations.